modules/core/04-channel/v2/keeper/packet.go

```diff
@@ -0,0 +1,78 @@
+package keeper
+
+import (
+	"context"
+	"strconv"
+
+	errorsmod "cosmossdk.io/errors"
+
+	clienttypes "github.com/cosmos/ibc-go/v9/modules/core/02-client/types"
+	channeltypes "github.com/cosmos/ibc-go/v9/modules/core/04-channel/types"
```

Comment:
Maybe we want to a new timeout type that lives just in channeltypesv2? To not need this dependency

Reply:
the timeout being referenced by the light client interface seems unavoidable to me, so I think that's the main consideration. Checkout https://github.com/cosmos/ibc-go/issues/7055 for context on how this might look going forward as well. In #7055, you could remove the timeout type and separate legacy behaviour by providing the timestamp/height type separately